File: configmodels/model_inventory.go
// SPDX-License-Identifier: Apache-2.0
// Copyright 2024 Canonical Ltd.

package configmodels

const (
	GnbDataColl = "webconsoleData.snapshots.gnbData"
	UpfDataColl = "webconsoleData.snapshots.upfData"
)

type Gnb struct {
	Name string `json:"name"`
	Tac  int32  `json:"tac,omitempty"`
}

type PostGnbRequest struct {
	Name string `json:"name"`
	Tac  int32  `json:"tac,omitempty"`
}

type PutGnbRequest struct {
	Tac int32 `json:"tac"`
}

type Upf struct {
	Hostname string `json:"hostname"`
	Port     string `json:"port"`
}

type PostUpfRequest struct {
	Hostname string `json:"hostname"`
	Port     string `json:"port"`
}

type PutUpfRequest struct {
	Port string `json:"port"`
}

File: ui/frontend_files/modules/objectsModels/model_inventory.js
/**
 * JavaScript equivalent of Go structs from model_inventory.go
 */

export const GNB_DATA_COLL = "webconsoleData.snapshots.gnbData";
export const UPF_DATA_COLL = "webconsoleData.snapshots.upfData";

export const Gnb = () => {
  return {
    name: "",
    tac: 0
  };
};

export const PostGnbRequest = () => {
  return {
    name: "",
    tac: 0
  };
};

export const PutGnbRequest = () => {
  return {
    tac: 0
  };
};

export const Upf = () => {
  return {
    hostname: "",
    port: ""
  };
};

export const PostUpfRequest = (hostname = "", port = "") => {
  return {
    hostname,
    port
  };
};

export const PutUpfRequest = () => {
  return {
    port: ""
  };
};
